js/base-tool-table.js

```diff
@@ -0,0 +1,393 @@
+function baseToolTable() {
+	
+	// Determines if a JSON is of the generated or TableExport format
+	function getFormat(t) {
+		if (t.colLabels) {
+			return "generated";
+		}
+		if (t.items) {
+			return "r20"
+		}
+	}
+	
+	// A function for generating and saving tables from JSONs of the generated format
+	function createGeneratedFormat(t) {
+		// Creates the table, with data for the full table
+		const r20t = d20.Campaign.rollabletables.create({
+			name: t.name.replace(/\s+/g, "-"),
+			id: d20plus.ut.generateRowId(),
+		});
+		
+		const labels = t.colLabels;
+		// Gets the index of the first column labeled with a dice roll
+		// For example, finds the d100 column
+		const dplace = labels.findIndex(l => /d[0-9]+/.test(l));
+		
+		r20t.tableitems.reset(t.rows.map(i => {
+			// Create the return value
+			const out = {
+				id: d20plus.ut.generateRowId(),
+				name: "",
+			};
+			
+			// Set the name
+			for (let col = 0; col < labels.length; col++) {
+				// Add a seperator for cases of multiple columns
+				if (out.name.length > 0) {
+					out.name += " | "
+				}
+				// Add each column to out.name
+				if (col !== dplace) {
+					// Get rid of ugly notation
+					const clean = Renderer.stripTags(i[col]);
+					out.name += clean;
+				}
+			}
+			
+			// Set the weight
+			if (dplace !== -1) {
+				const weight = i[dplace];
+				const dash = weight.indexOf("\u2013"); // Note: – is different from -
+				
+				// If the weight is a range
+				if (~dash) {
+					// Get the two numbers in the range, subtract them, add 1
+					const low = parseInt(weight.substring(0, dash));
+					let high = parseInt(weight.substring(dash + 1));
+					if (high === 0) high = 100;
+					out.weight = high - low + 1;
+				} else { // If the weight is a single value
+					out.weight = 1;
+				}
+			} else { // If the weight is unlisted
+				out.weight = 1;
+			}
+			
+			if (i.avatar) out.avatar = i.avatar;
+			return out;
+		}));
+		r20t.tableitems.forEach(it => it.save());
+	}
+	
+	// A function for generating and saving tables from JSONs of the Roll20 format
+	function createR20Format(t) {
+		const r20t = d20.Campaign.rollabletables.create({
+			name: t.name.replace(/\s+/g, "-"),
+			showplayers: t.isShown,
+			id: d20plus.ut.generateRowId(),
+		});
+		
+		r20t.tableitems.reset(t.items.map(i => {
+			const out = {
+				id: d20plus.ut.generateRowId(),
+				name: i.row,
+			};
+			if (i.weight !== undefined) out.weight = i.weight;
```

Comment:
prefer `!= null` over `!== undefined` (unless there's a specific reason to compare against _exactly_ `undefined` here?)